### DISCUSSION TITLE
Trying to delete Teleport App, will not work

### DISCUSSION THREAD

**Discussion by JosTheDude** [gravitational/teleport#28848] [Q&A] [answered]
I've been trying to delete an application from Teleport Applications TAB, but it simply will not delete despite me using the name. I've tried everything, how would I go about deleting it?

(I'm connected to our Teleport Auth Cluster)

![image](https://github.com/gravitational/teleport/assets/43651265/9475f513-38cd-47dd-ada5-3318679fea4f)

**Comment by webvictim** [comment, marked as answer]
Apps will stay visible in the cluster as long as there is at least one agent which is advertising them. When the agent goes offline or stops providing the app, it will disappear shortly afterwards (within 10 minutes at the most)

Given this, if you'd like to delete an app the way to do it is to modify the YAML configuration of the Teleport `app_service` on your `ns5005935` machine to remove that `pterodactyl-dev` app, then restart the Teleport process there. It should go away afterwards.
